mobile-app/app/screens/AppNavigator/screens/Balances/hooks/TokenLockedBalance.ts

```diff
@@ -0,0 +1,66 @@
+import { LoanVaultState } from '@defichain/whale-api-client/dist/api/loan'
+import { useWalletContext } from '@shared-contexts/WalletContext'
+import { useWhaleApiClient } from '@shared-contexts/WhaleContext'
+import { RootState } from '@store'
+import { fetchVaults, vaultsSelector } from '@store/loans'
+import BigNumber from 'bignumber.js'
+import { useCallback, useEffect, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+
+interface TokenLockedBalance {
+  [key: string]: BigNumber
+}
+
+export function useTokenLockedBalance ({ symbol }: { symbol: string }): BigNumber {
+  const vaults = useSelector((state: RootState) => vaultsSelector(state.loans))
+  const client = useWhaleApiClient()
+  const { address } = useWalletContext()
+  const dispatch = useDispatch()
+  const blockCount = useSelector((state: RootState) => state.block.count)
+  const [totalLockedBalance, setTotalLockedBalance] = useState<TokenLockedBalance>({})
+
+  useEffect(() => {
+    dispatch(fetchVaults({ client, address }))
+  }, [address, blockCount])
+
+  useEffect(() => {
+    setTotalLockedBalance(computeLockedAmount())
+  }, [vaults])
+
+  const computeLockedAmount = useCallback(() => {
+    const totalLockedBalance: TokenLockedBalance = {}
+
+    vaults.forEach(vault => {
+      if (vault.state === LoanVaultState.IN_LIQUIDATION) {
+        return
+      }
+
+      vault.collateralAmounts.forEach(collateral => {
+        const tokenExist = totalLockedBalance[collateral.symbol]
+        if (tokenExist !== undefined) {
+          totalLockedBalance[collateral.symbol] = totalLockedBalance[collateral.symbol].plus(collateral.amount)
+        } else {
+          totalLockedBalance[collateral.symbol] = new BigNumber(collateral.amount)
+        }
+      })
+    })
+
+    return totalLockedBalance
+  }, [vaults])
+
+  return totalLockedBalance[symbol] ?? new BigNumber(0)
```

Comment:
Can we return totalLockedBalance instead so that we can reuse it for calculating portfolio locked balance

Reply:
We can use a Map/Set for this. Then just sum up all the values of it

Reply:
We can make the symbol optional, if there's a symbol, return the amount of this token otherwise, return sum

Reply:
Or we can create a method which take symbol as parameter and gives specific value for the token

Reply:
updated to return sum